Lib9c/TableData/Summon/SummonSheet.cs

```diff
@@ -31,48 +49,20 @@ public override void Set(IReadOnlyList<string> fields)
                 CostNcg = ParseInt(fields[3]);
                 // Min. Two recipes are necessary
                 Recipes.Add((ParseInt(fields[4]), ParseInt(fields[5])));
-                CumulativeRecipe1Ratio = ParseInt(fields[5]);
                 Recipes.Add((ParseInt(fields[6]), ParseInt(fields[7])));
-                CumulativeRecipe2Ratio = CumulativeRecipe1Ratio + ParseInt(fields[7]);
-
-                // Recipe3 ~ 6 are optional
-                if (TryParseInt(fields[8], out _) && TryParseInt(fields[9], out _))
-                {
-                    Recipes.Add((ParseInt(fields[8]), ParseInt(fields[9])));
-                    CumulativeRecipe3Ratio = CumulativeRecipe2Ratio + ParseInt(fields[9]);
-                }
-                else
-                {
-                    CumulativeRecipe3Ratio = CumulativeRecipe2Ratio;
-                }
-
-                if (TryParseInt(fields[10], out _) && TryParseInt(fields[11], out _))
-                {
-                    Recipes.Add((ParseInt(fields[10]), ParseInt(fields[11])));
-                    CumulativeRecipe4Ratio = CumulativeRecipe3Ratio + ParseInt(fields[11]);
-                }
-                else
-                {
-                    CumulativeRecipe4Ratio = CumulativeRecipe3Ratio;
-                }
 
-                if (TryParseInt(fields[12], out _) && TryParseInt(fields[13], out _))
-                {
-                    Recipes.Add((ParseInt(fields[12]), ParseInt(fields[13])));
-                    CumulativeRecipe5Ratio = CumulativeRecipe4Ratio + ParseInt(fields[13]);
-                }
-                else
-                {
-                    CumulativeRecipe5Ratio = CumulativeRecipe4Ratio;
-                }
-                if (TryParseInt(fields[14], out _) && TryParseInt(fields[15], out _))
-                {
-                    Recipes.Add((ParseInt(fields[14]), ParseInt(fields[15])));
-                    CumulativeRecipe6Ratio = CumulativeRecipe5Ratio + ParseInt(fields[15]);
-                }
-                else
+                // Recipe3 ~ 15 are optional
+                for (var i = 3; i < 16; i++)
                 {
-                    CumulativeRecipe6Ratio = CumulativeRecipe5Ratio;
+                    var idx = 2 * i + 2;
+                    if (TryParseInt(fields[idx], out _) && TryParseInt(fields[idx + 1], out _))
+                    {
+                        Recipes.Add((ParseInt(fields[idx]), ParseInt(fields[idx+1])));
+                    }
+                    else
+                    {
+                        break;
+                    }
                 }
```

Comment:
Is it okay not to check `fields`' length? If not, `fields[idx]` may throw an error.

Reply:
It may use https://github.com/planetarium/lib9c/blob/development/Lib9c/TableCSV/Summon/SummonSheet.csv this CSV file, then it may not work as optional.